### PR TITLE
## OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by abiodunraheem**
## OOP school library: refactor your code

- Refactored Ruby code using SOLID principles.
- Applied Ruby best practices and language style guides in code.

### CODE REVIEW

**Review by OLIPLICHE — COMMENTED** [review]
Hi @abiodunraheem , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that shows that you did not work in pair-programming. I have checked the commit history. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -15,6 +15,72 @@ def list_persons(persons)
...
+
+  def person_student
+    puts 'Enter the name of the student'
+    name = gets.chomp
+    puts 'Enter the age of the student'
+    age = gets.chomp
+    classroom = 'Math'
+    new_student = Student.new(classroom, age, name)
+    create_person(@person, new_student)
+    puts "Student #{name} has been added to the library"
```
Comment (zhadier) [inline app.rb]: - [ ] In all of your methods in the `App.rb` file, you're both getting `an input from users` and then `creating an object` using that input. This method does not follow the single responsibility principle as a method should only do 1, ask for an input from the user or create an object. Please separate these operations into different methods to achieve the `single responsibility principle`. You may have a separate class for validating and getting inputs if you want. Kindly do this for all your methods in App.rb.

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @abiodunraheem & @GOlivierNation  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ GitFlow was set up correctly
✔️ Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @GOlivierNation, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------

- _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**